contracts/tokenomics/generator/src/contract.rs

```diff
@@ -1197,6 +1201,74 @@ fn send_orphan_proxy_rewards(
         .add_attribute("amount", amount))
 }
 
+/// ## Description
+/// Sets the reward proxy contract for the pool. Returns an [`ContractError`] on failure, otherwise
+/// returns the [`Response`] with the specified attributes if the operation was successful.
+fn update_pool_config(
+    deps: DepsMut,
+    lp_token: String,
+    proxy: String,
+) -> Result<Response, ContractError> {
+    let lp_addr = addr_validate_to_lower(deps.api, &lp_token)?;
+    let proxy_addr = addr_validate_to_lower(deps.api, &proxy)?;
+
+    let mut pool_info = POOL_INFO.load(deps.storage, &lp_addr)?;
+    pool_info.reward_proxy = Some(proxy_addr);
```

Comment:
just setting the reward_proxy is not enough here, there's a whole logic that depends on whether there is a proxy. check add function to see it